### PR TITLE
UXUI-239 Icon list component

### PR DESCRIPTION
**Pull request by jackdcrawford**
#### What does this PR do? (please provide any background)

We've seen a common ui pattern crop up where we have a list that uses icons for the bullets. I've created `IconList` & `IconListItem` to make this UI pattern easier.

Necessary for [Jira](https://holidayextras.jira.com/browse/UXUI-239)

Other tweaks:
- Supervisor not required for docs as statically served
- Catch errors on docs server start
- Fix file resource error happening when watch task is running

#### What tests does this PR have?

Unit tests for both new components

#### How can be this tested?

Pull this branch, `npm install` & `grunt docs`.

Visit [http://localhost:4000/#lists](http://localhost:4000/#lists)

###### Toolkit component
toolkit

###### Example on the Holiday Extras site
HX

#### What gif best describes how you feel about this work?
lulz

---
#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):
- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.

#### Software Engineer or Developer review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve run all the tests locally and they pass.

#### Software Engineer or project guru review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve run all the tests locally and they pass.

### CODE REVIEW

**Comment by nickine** [comment]
Shall take the dev

**Comment by andrewcashmore** [comment]
Issue with font awesome not pulling in.

**Comment by jackdcrawford** [comment]
@andrewcashmore oversight from me as I have Font Awesome installed locally. I'll find a way to address this, thanks for picking it up.

**Comment by jackdcrawford** [comment]
cc @joestump

**Comment by joestump** [comment]
There appears to be some button related mods in this PR? Not sure what those are about, but the item list component stuff looks good.

File: src/components/button/code/less/_button.less [hunk 1]
```diff
@@ -44,17 +44,25 @@
...
-.component-button {
+  //
+  // == Block ==
+  //
   &.block {
     display:block;
     width: 100%;
   }
+
+  // If a button horizontally follows another button, put a left margin
```
Comment (jodiedoubleday) [inline src/components/button/code/less/_button.less]: Really like this, great idea!

**Comment by nickine** [comment]
Everything looks good to me

**Comment by HeinBehrenshx** [comment]
Taking se

**Comment by HeinBehrenshx** [comment]
+1

**Pull request merged**